### PR TITLE
[2.0] Add more context to create diagnosticsts error.

### PR DESCRIPTION
**Pull request by janisz**
Backports #384
---

* Add more context to create diagnosticsts error.
* Run linter on tests
* Use go structs instead of raw JSON in list test.

Fixes: https://jira.mesosphere.com/browse/COPS-5439

### CODE REVIEW

**Comment by janisz** [comment]
@mesosphere-ci run windows integration tests

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 * Fixes
 
+  * Add better erros descriptions for `nodes diagnostics` command.
```
Comment (bamarni) [inline CHANGELOG.md]:
```suggestion
  * Add better error descriptions for `node diagnostics` command.
```

**Review by bamarni — APPROVED** [review]

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

**Review by makkes — APPROVED** [review]

**Pull request merged**